### PR TITLE
docs: v1 - diataxis framework

### PR DESCRIPTION
**Pull request by andynog**
close: #1613 

New structure for documentation using the [diataxis framework](https://diataxis.fr). Created landing pages for the major sections and re-organized the existing content into one of the 4 topics of the diataxis framework (tutorials, how-to guides, technical reference and explanation)

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [X] Updated relevant documentation (`docs/` or `spec/`) and code comments
- [ ] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -16,14 +16,14 @@ CometBFT serves blockchain applications.
 
 More formally, CometBFT performs Byzantine Fault Tolerant (BFT)
 State Machine Replication (SMR) for arbitrary deterministic, finite state machines.
-For more background, see [What is CometBFT?](introduction/README.md#what-is-cometbft.md).
+For more background, see [What is CometBFT?](explanation/introduction/README.md#what-is-cometbft.md).
```
Comment (melekes) [inline docs/README.md:19]: 404 locally

File: docs/README.md [hunk 2]
```diff
@@ -16,14 +16,14 @@ CometBFT serves blockchain applications.
...
 
 To get started quickly with an example application, see the [quick start guide](guides/quick-start.md).
```
Comment (melekes) [inline docs/README.md:21]: 404 locally

File: docs/README.md [hunk 3]
```diff
@@ -16,14 +16,14 @@ CometBFT serves blockchain applications.
...
 
 To learn about application development on CometBFT, see the [Application Blockchain Interface](https://github.com/cometbft/cometbft/tree/main/spec/abci).
 
 For more details on using CometBFT, see the respective documentation for
-[CometBFT internals](core/), [benchmarking and monitoring](tools/), and [network deployments](networks/).
+[CometBFT internals](explanation/core/), [benchmarking and monitoring](guides/tools/), and [network deployments](networks/).
```
Comment (melekes) [inline docs/README.md:26]: 404 locally for `[network deployments](networks/)`

**Review by melekes — COMMENTED** [review]
massive 🚀

**Comment by melekes** [comment]
Maybe we should also check for 404 errors with a tool like https://github.com/stevenvachon/broken-link-checker

**Comment by andynog** [comment]
> Maybe we should also check for 404 errors with a tool like https://github.com/stevenvachon/broken-link-checker

Hi @melekes, we do have a broken links check in the repo, I'll run that again to ensure we are not missing anything. The tool is not perfect because it gives false positives especially in the # anchor links, but overall it is helpful.

File: docs/README.md (hunk 2 above)
Comment (andynog) [reply docs/README.md]: fixed

File: docs/README.md (hunk 3 above)
Comment (andynog) [reply docs/README.md]: fixed

File: docs/README.md (hunk 1 above)
Comment (andynog) [reply docs/README.md]: Fixed. Thanks !

**Review by melekes — APPROVED** [review]
👍

**Review by adizere — APPROVED** [review]
Thanks for the huge lift here Andy, LGTM!

**Comment by andynog** [comment]
Thanks @adizere and @melekes

**Pull request merged**